cli/src/pcluster/validators/cluster_validators.py

```diff
@@ -184,6 +185,27 @@ def _validate(self, efa_enabled: bool, os: str, architecture: str):
             )
 
 
+class SchedulableMemoryValidator(Validator):
+    """Validate SchedulableMemory parameter passed by user."""
+
+    def _validate(self, schedulable_memory, ec2memory, instance_type):
+        if schedulable_memory is not None:
+            if schedulable_memory < 1:
+                self._add_failure("SchedulableMemory must be at least 1 MiB.", FailureLevel.ERROR)
+            if schedulable_memory > ec2memory:
+                self._add_failure(
+                    f"SchedulableMemory cannot be larger than EC2 Memory for selected instance type "
+                    f"{instance_type} ({ec2memory} MiB).",
+                    FailureLevel.ERROR,
+                )
+            if schedulable_memory < math.floor(0.95 * ec2memory):
+                self._add_failure(
+                    f"SchedulableMemory was set lower than 95% of EC2 Memory for selected instance type "
+                    f"{instance_type} ({ec2memory} MiB).",
+                    FailureLevel.WARNING,
+                )
```

Comment:
Overriding SchedulableMemory is already an advanced setting that I'd expect users to do if they want to change our default even to decrease it so probably I'd skip this warning for now.

Reply:
I think this point was raised by the PM in the design doc.

Reply:
Lowered FailureLevel to INFO.